### PR TITLE
Push/Pull return the descriptor of the associated tag

### LINKED ISSUE #68: Optionally return the digest of the manifest
Optionally return the digest of the manifest on push and pull.

### PR DESCRIPTION
**Pull request by shizhMSFT**
The public APIs of the `oras` pacakge are changed to
```go
func Pull(ctx context.Context, resolver remotes.Resolver, ref string, ingester content.Ingester, opts ...PullOpt) (ocispec.Descriptor, []ocispec.Descriptor, error) {}
func Push(ctx context.Context, resolver remotes.Resolver, ref string, provider content.Provider, descriptors []ocispec.Descriptor, opts ...PushOpt) (ocispec.Descriptor, error) {}
```
@jdolitsky you need to send a PR to **helm/helm** to update the oras API once this PR is merged.

By default, `oras` displays nothing. With `-v` verbose option, `oras` shows
```
$ oras push -v localhost:5000/hello:latest hi.txt
hi.txt
Pushed localhost:5000/hello:latest
sha256:0ee289c9702eb28bf59b0625462daa632c525445a31e606f36c8900f50c86192
$ oras pull -v localhost:5000/hello:latest
WARN[0000] unknown_type: application/vnd.oci.image.config.v1+json
hi.txt
Pulled localhost:5000/hello:latest
sha256:0ee289c9702eb28bf59b0625462daa632c525445a31e606f36c8900f50c86192
```

Resolves #68 
Closes #67 as this PR based on it.

### CODE REVIEW

File: cmd/oras/pull.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ func runPull(opts pullOptions) error {
 				fmt.Println(name)
 			}
 		}
+		fmt.Printf("pulled %s: %s\n", opts.targetRef, desc.Digest)
```
Comment (sajayantony) [inline cmd/oras/pull.go]: Would recommend Capitalizing the message start. 
Also maybe break up the digest into another line. 

For reference this is how docker outputs the messages: 
```
Using default tag: latest
latest: Pulling from library/hello-world
Digest: sha256:92695bc579f31df7a63da6922075d0666e565ceccad16b59c3374d2cf4e8e50e
Status: Image is up to date for hello-world:latest
```

File: cmd/oras/push.go [hunk 2]
```diff
@@ -132,7 +138,15 @@ func runPush(opts pushOptions) error {
...
 	// ready to push
 	resolver := newResolver(opts.username, opts.password, opts.configs...)
-	return oras.Push(context.Background(), resolver, opts.targetRef, store, files, pushOpts...)
+	desc, err := oras.Push(context.Background(), resolver, opts.targetRef, store, files, pushOpts...)
+	if err != nil {
+		return err
+	}
+
+	if opts.verbose {
+		fmt.Printf("pushed %s: %s\n", opts.targetRef, desc.Digest)
```
Comment (sajayantony) [inline cmd/oras/push.go]: NIT: Similar to the previous comment. Capitalize the start. 
Also is push message in verbose while pull message isn't?

File: pkg/oras/oras_test.go [hunk 3]
```diff
@@ -160,7 +161,7 @@ func (suite *ORASTestSuite) Test_1_Pull() {
 	// Pull chart-dir
 	store = orascontent.NewMemoryStore()
 	ref = fmt.Sprintf("%s/chart-dir:test", suite.DockerRegistryHost)
-	descriptors, err = Pull(newContext(), newResolver(), ref, store)
+	_, descriptors, err = Pull(newContext(), newResolver(), ref, store)
```
Comment (sajayantony) [inline pkg/oras/oras_test.go]: Do you have a test for `WithAllowedMediaType` ?

File: cmd/oras/push.go (hunk 2 above)
Comment (shizhMSFT) [reply cmd/oras/push.go]: Both are in verbose

File: pkg/oras/oras_test.go (hunk 3 above)
Comment (shizhMSFT) [reply pkg/oras/oras_test.go]: will add one

Comment (shizhMSFT) [reply pkg/oras/oras_test.go]: added test `Test_2_MediaType`

**Pull request merged**